crates/vfs/src/lib.rs

```diff
@@ -137,13 +139,12 @@ impl Vfs {
     ///
     /// This will skip deleted files.
     pub fn iter(&self) -> impl Iterator<Item = (FileId, &VfsPath)> + '_ {
-        (0..self.data.len())
-            .map(|it| FileId(it as u32))
-            .filter(move |&file_id| self.get(file_id).is_some())
-            .map(move |file_id| {
-                let path = self.interner.lookup(file_id);
-                (file_id, path)
-            })
+        (0..self.data.len()).filter_map(move |it| {
+            let file_id = FileId(it as u32);
+            let _ = self.get(file_id).as_ref()?;
+            let path = self.interner.lookup(file_id);
+            Some((file_id, path))
+        })
```

Comment:
The original formulation is intentional: every `.filter(...).map(...)` sequence can be re-written as `.filter_map`, but converse is not true -- if `filter_map` "changes the types", then it can't be replaced with separate `filter` and `map`. 

That's why, when it is possible to not use `filter_map`, I avoid using it: using less powerful construct makes the code more readable, because there's less things that the code can do.